src/Oscoin/Crypto/PubKey/FileStore.hs

```diff
@@ -0,0 +1,88 @@
+-- | This module provides the 'writeKeyPair' and 'readKeyPair'
+-- functions to read and write real world 'Crypto' keypairs to disk.
+module Oscoin.Crypto.PubKey.FileStore
+    ( writeKeyPair
+    , readKeyPair
+    , ReadKeyError(..)
+    ) where
+
+import           Oscoin.Prelude
+
+import           Oscoin.Crypto (Crypto)
+import qualified Oscoin.Crypto.PubKey as Crypto
+import qualified Oscoin.Crypto.PubKey.RealWorld as Crypto.RealWorld
+
+import           Codec.Serialise
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as LBS
+import           System.Directory
+import           System.FilePath
+import           System.IO.Error (isDoesNotExistError)
+
+
+-- | Read a keypair from disk. See 'writeKeyPair' about how the path
+-- parameter determines the key location. May raise 'ReadKeyError'.
+readKeyPair :: (MonadIO m) => FilePath -> m (Crypto.KeyPair Crypto)
+readKeyPair keysPathPrefix = liftIO $ do
+    let (pkPath, skPath) = keyPaths keysPathPrefix
+    pk <- readKeyFile deserialiseOrFail pkPath
+    sk <- readKeyFile Crypto.RealWorld.deserialisePrivateKey skPath
+    pure (pk ,sk)
+  where
+    readKeyFile customDeserialiseOrFail path = do
```

Comment:
Should this live at the top-level? Or what's the idea behind passing in the deserialiser?

Reply:
The custom deserializer is passed out of necessetiy since we don’t have a `Serialise` instance for the private key. I don’t think it needs to be a top-level function. Ideally the signature of this would be
~~~haskell
readKeyFile :: (Serialise a) => FilePath -> IO a
~~~
but this fails because there is no `Serialise PrivateKey` instance. The `readKeyFile` hacks around this by accepting the serialization as an argument.